### DISCUSSION TITLE
我已在虚拟环境安装python的第三方库 但是运行nb时提示 No module named

### DISCUSSION THREAD

**Discussion by Ah-Q** [nonebot/discussions#449] [问答] [answered]
虚拟环境已经安装了python的第三方库 但是运行nb的时候依旧提示“No module named ***”，请问是怎么回事呢

**Comment by yanyongyu** [comment]
你确定在nb使用的虚拟环境中安装了这个库吗？你可以通过`pip list`来查看已安装的包列表`pip -V`来查看当前环境路径

**Reply by Ah-Q** [reply]
是的 我已在虚拟环境中安装了此三方库（我使用的虚拟环境是venv ）但是奇怪的是我无法再全局环境中安装三方库（全局环境中使用命令 pip -V，显示的环境路径是虚拟环境的路径）

**Reply by yanyongyu** [reply]
运行nb的环境和你安装库使用的虚拟环境不是一个环境？

**Reply by Ah-Q** [reply]
是同一个 我是先创建的虚拟环境venv然后在此环境中部署的nb

**Reply by yanyongyu** [reply]
请提供你的报错截图以及对应环境的pip list

**Reply by Ah-Q** [reply]
虚拟环境中的pi'ppip list结果：
![image](https://github.com/nonebot/discussions/assets/80017370/ab14a39d-9426-499f-823c-161057485713)
![image](https://github.com/nonebot/discussions/assets/80017370/4b9fee31-5e88-4b1d-a8ec-9ec660194cc9)
报错截图：
![image](https://github.com/nonebot/discussions/assets/80017370/20f9316c-e3d1-4b96-9d71-01e1c270b45d)

**Reply by Ah-Q** [reply]
nb是在screen会话中运行的 也许您需要知道这个

**Reply by yanyongyu** [reply]
看上去你的激活的虚拟环境并不对啊，你用`/root/Lvenv/bin/LBot-2810/.venv/bin/pip list`看看

**Reply by Ah-Q** [reply]
此命令结果中没有requests这个第三方库
![image](https://github.com/nonebot/discussions/assets/80017370/5cc75d1a-83e2-46f0-9476-4f1f30ee4e07)

**Reply by Ah-Q** [reply]
这个结果不是在虚拟环境中运行的

**Reply by yanyongyu** [reply]
所以是你搞错了虚拟环境，你应该使用 /root/Lvenv/bin/LBot-2810/.venv 这个虚拟环境

**Reply by Ah-Q** [reply]
当我将`/root/Lvenv/lib/python3.9/site-packages/requests`复制到`/root/Lvenv/bin/LBot-2810/.venv/lib/python3.9/site-packages`时
提示的错误为：
![image](https://github.com/nonebot/discussions/assets/80017370/97c92c2e-d360-44d8-9640-43608a21940a)

**Reply by yanyongyu** [reply]
。。。哪有用复制的，你需要激活虚拟环境然后使用 pip 安装库，你可以参考 python 官方文档使用虚拟环境

**Reply by Ah-Q** [reply]
` /root/Lvenv/bin/LBot-2810/.venv`是创建nb时生成的文件 请问我应该如何使用这个虚拟环境

**Reply by Ah-Q** [reply]
或者说我该如何更换启动nb时加载的python库呢

**Reply by Ah-Q** [reply]
可是我本身就已经是在我创建的虚拟环境中了 我用的是venv

**Reply by yanyongyu** [reply, marked as answer]
nb已经为你的bot项目创建了venv，你应该用nb的venv来安装库，你可以使用 `source /root/Lvenv/bin/LBot-2810/.venv/bin/activate` 来激活虚拟环境

**Reply by Ah-Q** [reply]
谢谢你
